src/validator.cc

```diff
@@ -1152,16 +1152,21 @@ void Validator::CheckTable(const Location* loc, const Table* table) {
 }
 
 void Validator::CheckElemSegments(const Module* module) {
+  bool bulk_or_ref_enabled = options_.features.bulk_memory_enabled() ||
+                             options_.features.reference_types_enabled();
+
   for (const ModuleField& field : module->fields) {
     if (auto elem_segment_field = dyn_cast<ElemSegmentModuleField>(&field)) {
       auto&& elem_segment = elem_segment_field->elem_segment;
       for (const ElemExpr& elem_expr : elem_segment.elem_exprs) {
         if (elem_expr.kind == ElemExprKind::RefFunc) {
           CheckFuncVar(&elem_expr.var, nullptr);
+        } else if (!bulk_or_ref_enabled) {
+          PrintError(&field.loc, "ref.null is not allowed");
```

Comment:
What if `elem_expr.kind` is not `RefNul`?

Reply:
The enum only allows RefNull and RefFunc for now, so if it were something else there's an error earlier in the pipeline.

Reply:
I still rather see an explicit check here followed by an "else { UNREACHABLE }" to catch the other reftypes when they start to make sense.

Reply:
Sorry, responded before pushing. :)

Reply:
Whoops, thought you were talking about something else. :)